fuel/datasets/hdf5.py

```diff
@@ -326,35 +360,55 @@ def _file_handle(self):
 
     def get_data(self, state=None, request=None):
         if self.load_in_memory:
-            return self._in_memory_get_data(state=state, request=request)
+            data, data_shapes = self._in_memory_get_data(state, request)
         else:
-            return self._out_of_memory_get_data(state=state, request=request)
+            data, data_shapes = self._out_of_memory_get_data(state, request)
+        for i, shapes in enumerate(data_shapes):
```

Comment:
That's much better. I would also get rid of the intermediate variable `shapes`, because the difference between `shapes` and `data_shapes` is not immediately obvious. You can call old `data_shapes` `shapes`, and instead of old `shapes` simply use `shapes[i]`. I would even go further to say that mixture of `zip` and `enumerate` looks a bit weird, and maybe it would be better to use only indices here, i.e. `data[i][j] = data[i][j].reshape(shapes[i][j])`.